src/thor/costmatrix.cc

```diff
@@ -78,7 +78,13 @@ float CostMatrix::GetCostThreshold(const float max_matrix_distance) {
 // Clear the temporary information generated during time + distance matrix
 // construction.
 void CostMatrix::clear() {
-  // Clear the target edge markings
+  // Clear the source/target edge markings
+  for (auto& iter : *sources_) {
+    iter.second.clear();
+    iter.second.resize(0);
+    iter.second.shrink_to_fit();
+  }
```

Comment:
this, and the same version below for targets is definitely not free. here we pay the cost of dealloacting everything and then in the next request we have to do the allocations all over again. it would be interesting here to instead do something like:

```suggestion
  for (auto& iter : *sources_) {
    iter.second.resize(MATRIX_CONNECTION_ENTRIES_MAX_BUFFER_SIZE);
    iter.second.shrink_to_fit();
    iter.second.clear();
  }
```

and the same below for targets

Reply:
yeah, seems like I missed a spot there!